anta/tests/system.py

```diff
@@ -299,3 +303,93 @@ def test(self) -> None:
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the Primary NTP server (marked as preferred) has the condition 'sys.peer' and
+    all other NTP servers have the condition 'candidate'.
+    * Failure: The test will fail if the Primary NTP server (marked as preferred) does not have the condition 'sys.peer' or
+    if any other NTP server does not have the condition 'candidate'.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.system:
+      - VerifyNTPAssociations:
+          ntp_servers:
+            - server_address: 1.1.1.1
+              preferred: True
+              stratum: 1
+            - server_address: 2.2.2.2
+              stratum: 2
+            - server_address: 3.3.3.3
+              stratum: 2
+    ```
+    """
+
+    name = "VerifyNTPAssociations"
+    description = "Verifies the Network Time Protocol (NTP) associations."
+    categories: ClassVar[list[str]] = ["system"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show ntp associations")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyNTPAssociations test."""
+
+        ntp_servers: list[NTPServer]
+        """List of NTP servers."""
+
+        class NTPServer(BaseModel):
+            """Model for a NTP server."""
+
+            server_address: str | IPv4Address
+            """The NTP server address as an IPv4 address or hostname. The NTP server name defined in the running configuration
+            of the device may change during DNS resolution, which is not handled in ANTA. Please provide the DNS-resolved server name.
+            For example, 'ntp.example.com' in the configuration might resolve to 'ntp3.example.com' in the device output."""
+            preferred: bool = False
+            """Optional preferred for NTP server. If not provided, it defaults to `False`."""
+            stratum: int = Field(ge=0, le=16)
+            """NTP stratum level (0 to 15) where 0 is the reference clock and 16 indicates unsynchronized.
+            Values should be between 0 and 15 for valid synchronization and 16 represents an out-of-sync state."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyNTPAssociations."""
+        failures: str = ""
+
+        if not (peer_details := get_value(self.instance_commands[0].json_output, "peers")):
+            self.result.is_failure("None of NTP peers are not configured.")
+            return
+
+        # Iterate over each NTP server.
+        for ntp_server in self.inputs.ntp_servers:
+            server_address = str(ntp_server.server_address)
+            preferred = ntp_server.preferred
+            stratum = ntp_server.stratum
+
+            # Check if NTP server details exists.
+            if (peer_detail := get_value(peer_details, server_address, separator="..")) is None:
+                failures += f"\nNTP peer {server_address} is not configured."
```

Comment:
I don't like having unnecessary carriage returns in the result messages. When there is only 1 message:
![image](https://github.com/user-attachments/assets/f362c09a-8a5f-4af8-8a09-66b546e9c736)

Reply:
Removed the unnecessary \n at start. Thanks!!